elyra/contents/file_parser.py

```diff
@@ -0,0 +1,279 @@
+class PythonFileParser(FileParser):
+
+    _type = "python"
+
+    '''
+    def __init__(self):
+        # super().__init__(operation_filepath)
+    '''
+
+    def type(self):
+        return self._type
+
+    def file_as_list_of_code_chunks(self, filepath) -> List[str]:
+        with open(filepath) as f:
+            code_chunks = f.readlines()
+        return [c.strip() for c in code_chunks]
+
+    def get_next_code_chunk(self, filepath):
+        with open(filepath) as f:
+            for line in f:
+                yield line.strip()
+
+    def get_resources(self, filepath):
+        model = dict()
+        model["env_list"] = dict()
+        model["inputs"] = dict()
+        model["outputs"] = dict()
+
+        for line in self.get_next_code_chunk(filepath):
+            if line:
+                matches = PythonScriptParser().parse_environment_variables(line)
+                for key, match in matches:
+                    model[key][match.group(1)] = match.group(2)
+
+        return model
+
+
+class RFileParser(FileParser):
+
+    _type = "r"
+
+    '''
+    def __init__(self):
+        # super().__init__(operation_filepath)
+    '''
+
+    def type(self):
+        return self._type
+
+    def file_as_list_of_code_chunks(self, filepath) -> List[str]:
+        with open(filepath) as f:
+            code_chunks = f.readlines()
+        return [c.strip() for c in code_chunks]
+
+    def get_next_code_chunk(self, filepath):
+        with open(filepath) as f:
+            for line in f:
+                yield line.strip()
+
+    def get_resources(self, filepath):
+        model = dict()
+        model["env_list"] = dict()
+        model["inputs"] = dict()
+        model["outputs"] = dict()
+
+        for line in self.get_next_code_chunk(filepath):
+            if line:
+                matches = RScriptParser().parse_environment_variables(line)
+                for key, match in matches:
+                    model[key][match.group(1)] = match.group(2)
+
+        return model
+
+
+class PythonScriptParser:
+    def search_expressions(self) -> List:
+        # TODO: add more key:list-of-regex pairs to parse for additional resources
+        regex_dict = dict()
+
+        # First regex matches envvar assignments of form os.environ["name"] = value
+        # Second regex matches envvar assignments that use os.getenv("name", "value") with default provided
+        # Third regex matches envvar assignments that use os.environ.get("name", "value") with default provided
+        # Both name and value are captured if possible
+        envs = [r"os\.environ\[[\"']([a-zA-Z_]+[A-Za-z0-9_]*)[\"']\]\s*=(?:\s*[\"'](.[^\"']*)?[\"'])?",
+                r"os\.getenv\([\"']([a-zA-Z_]+[A-Za-z0-9_]*)[\"']\s*\,(?:\s*[\"'](.[^\"']*)?[\"'])?",
+                r"os\.environ\.get\([\"']([a-zA-Z_]+[A-Za-z0-9_]*)[\"']\s*\,(?:\s*[\"'](.[^\"']*)?[\"'])?"]
+        regex_dict["env_list"] = envs
+        return regex_dict
+
+    def parse_environment_variables(self, line):
```

Comment:
Let's move this method into a `ScriptParser` base class from which both Python and R script parsers derive.